pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/TransactionAggregator.java

```diff
@@ -0,0 +1,292 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.stats.prometheus;
+
+import io.jsonwebtoken.io.IOException;
+import io.netty.util.concurrent.FastThreadLocal;
+import org.apache.bookkeeper.mledger.ManagedLedger;
+import org.apache.bookkeeper.mledger.impl.ManagedLedgerMBeanImpl;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.broker.service.persistent.PersistentSubscription;
+import org.apache.pulsar.broker.service.persistent.PersistentTopic;
+import org.apache.pulsar.common.naming.NamespaceName;
+import org.apache.pulsar.common.util.SimpleTextOutputStream;
+import org.apache.pulsar.transaction.coordinator.impl.MLTransactionLogImpl;
+import org.apache.pulsar.transaction.coordinator.impl.MLTransactionMetadataStore;
+import org.apache.pulsar.transaction.coordinator.impl.TransactionMetadataStoreStats;
+
+public class TransactionAggregator {
+
+    private final static FastThreadLocal<AggregatedTransactionCoordinatorStats> localTransactionCoordinatorStats =
+            new FastThreadLocal<AggregatedTransactionCoordinatorStats>() {
+                @Override
+                protected AggregatedTransactionCoordinatorStats initialValue() throws Exception {
+                    return new AggregatedTransactionCoordinatorStats();
+                }
+            };
+
+    private final static FastThreadLocal<ManagedLedgerStats> localManageLedgerStats =
+            new FastThreadLocal<ManagedLedgerStats>() {
+                @Override
+                protected ManagedLedgerStats initialValue() throws Exception {
+                    return new ManagedLedgerStats();
+                }
+            };
+
+    public static void generate(PulsarService pulsar, SimpleTextOutputStream stream) {
+        String cluster = pulsar.getConfiguration().getClusterName();
+        pulsar.getBrokerService().getMultiLayerTopicMap().forEach((namespace, bundlesMap) -> {
+
+            bundlesMap.forEach((bundle, topicsMap) -> {
```

Comment:
Make sure the managed ledger stats of the pending ack and transaction log only been exposed when exposeTopicMetricsEnabled.